js/components/graph/Graph.js

```diff
@@ -422,6 +422,21 @@ export class Graph extends React.Component {
     }
   }
 
+  nodeUnselect = courseId => {
+    const courseLabelArray = this.state.nodesJSON[courseId].text
+    const courseLabel = courseLabelArray[courseLabelArray.length - 1].text
+    const wasSelected = this.state.nodesStatus[courseId].selected
+    const temp = this.state.selectedNodes
+
+    wasSelected ? this.toggleSelection(courseId) : {}
```

Comment:
In general, don't use ternary ifs with side effects; use if statements instead.

In this specific case, combine this with the if statement below.